src/nanoemoji/util.py

```diff
@@ -85,30 +87,120 @@ def file_printer(filename):
             yield partial(print, file=f)
 
 
-def ensure_ttfont_fully_decompiled(ttfont: ttLib.TTFont):
-    # A TTFont might be opened lazily and some tables only partially decompiled.
-    # So for this to work on any TTFont, we first compile everything to a temporary
-    # stream then reload with lazy=False. Input font is modified in-place.
+def _assert_all_keys_loaded(font: ttLib.TTFont):
+    not_loaded = sorted(t for t in font.keys() if not font.isLoaded(t))
+    assert (
+        not not_loaded
+    ), f"Everything should be loaded, following aren't: {not_loaded}"
+
+
+def _reload(font: ttLib.TTFont, lazy: bool = True):
+    # Stream font to memory and load it back again
     tmp = BytesIO()
-    ttfont.save(tmp)
+    font.save(tmp)
     tmp.seek(0)
-    ttfont2 = ttLib.TTFont(tmp, lazy=False)
-    for tag in ttfont2.keys():
-        table = ttfont2[tag]
+    font2 = ttLib.TTFont(tmp, lazy=lazy)
+
+
+def load_fully(font: Union[Path, ttLib.TTFont]) -> ttLib.TTFont:
+    if isinstance(font, Path):
+        font = ttLib.TTFont(str(font), lazy=False)
+    else:
+        # A TTFont might be opened lazily and some tables only partially decompiled.
+        # If so, reload it
+        if font.lazy:
```

Comment:
`lazy` has three states, True, False and None (which is the default and means something in between). I know, very confusing; it should be documented better.
```suggestion
        if font.lazy is not False:
```

Reply:
note that my old `ensure_ttfont_fully_decompiled` was called unconditionally before, not just when lazy. The rationale was that the TTFont we receive from ufo2ft might be in an in-between state that is resolved only upon serializing the font (to disk or memory) because something else happens or is computed upon compile, e.g. the post table format may be set to 3 to drop the glyph names but the rest of the loaded tables still use the old names and it's only upon compile that the names are actually stripped away; so we'd compile the font to a temporary stream and reload off that.

Reply:
it's not just "A TTFont might be opened lazily and some tables only partially decompiled." As I explained in my previous comment, it's also possible that the TTFont object is in an in-between state preceding the compile step during which some stuff may get updated or flushed away, think about post v3 glyph names dropped or glyph bounding boxes recalcuated - none of these are not related to the lazy flag.
I think you should either run `_reload` unconditionally irrespective of font.lazy (if we want to support that in-between case) or we don't run at all and simply rely on the new ensureDecompiled() to make sure that any tables that have not been loaded yet will be. Running _reload only for font.lazy = True is either unnecessary or incomplete.

Reply:
i think you can remove the _reload altogether and if we do find issues we address them

Reply:
the "A TTFont might be opened lazily and some tables only partially decompiled" is already taken care of by the subsequent `ensureDecompiled()`